### PR TITLE
SocketIO update to 4.5.0

### PR DESCRIPTION
**Pull request by Arri98**
Updated socketIO to version 4.5.0 as the 2.4.0 was deprecated. For this the google-closure-compiler had to be updated as the new version of SocketIO used JS features that the compiler didn't support.

[Yes] It needs and includes Unit Tests

Updated test to account for changes in the socketIO API.

[] It includes documentation for these changes in `/doc`.

### CODE REVIEW

**Review by xmartinez — COMMENTED** [review]
@Arri98 Looking good :-)

We should update package lock files, too, but it might be better to wait until #1818 has been merged, so we can base this PR on the new lock file format.

I noticed that `spine` also uses the `socket.io-client` library. It might make sense to update it also in this PR (so that perftest are in sync with erizo).

**Comment by xmartinez** [comment]
@Arri98 @lodoyun We have been testing this change (rebased on the nodejs upgrade, PR #1817) for a while.

We have performed a load test using a single stream with up to 1200 concurrent subscribers:

Screenshot from 2022-07-28 18-34-16

and have observed no signalling issues.

We have also been dogfooding the change in our internal environment, which exposes it to a higher variety of browser/OS combinations (Win/Linux/Mac, Chrome/Firefox/Safari). No issues have been observed, either.

**Comment by xmartinez** [comment]
@Arri98 @lodoyun We have detected some issues related to these upgrade on our staging and prod environments. We are still investigating the problem. I will comment on this PR once we have more details.

Until we root cause and fix the problem, we do not recommend merging this PR.

**Comment by Arri98** [comment]
Some events (https://socket.io/docs/v4/emit-cheatsheet/) and event error handling (https://socket.io/docs/v3/migrating-from-2-x-to-3-0/#a-middleware-error-will-now-emit-an-error-object) changed from v2 to v3 and caused issues when a socket disconnected. I will update this PR after solving the issues.

File: erizo_controller/erizoClient/gulp/erizoTasks.js [hunk 1]
```diff
@@ -24,8 +24,8 @@ const erizoTasks = (gulp, plugins, config) => {
       .pipe(plugins.sourcemaps.init())
       .pipe(plugins.closureCompiler({
         warning_level: 'QUIET',
-        languageIn: 'ECMASCRIPT6',
-        languageOut: 'ECMASCRIPT5',
+        languageIn: 'ECMASCRIPT_2017',
```
Comment (lodoyun) [inline erizo_controller/erizoClient/gulp/erizoTasks.js:27]: 👍

File: erizo_controller/erizoClient/src/Socket.js [hunk 2]
```diff
@@ -133,21 +132,16 @@ const Socket = (newIo) => {
       }
     });
 
-    reliableSocket.on('error', (err) => {
-      log.warning(`message: socket error, id: ${that.id}, state: ${that.state.toString()}, error: ${err}`);
+    reliableSocket.on('connect_error', (err) => {
+      // Fired when an namespace middleware error occurs.
+      log.warning(`message: connect_error error, id: ${that.id}, state: ${that.state.toString()}, error: ${err}`);
```
Comment (lodoyun) [inline erizo_controller/erizoClient/src/Socket.js:137]: 👍

**Review by lodoyun — APPROVED** [review]
This looks good to me 👍

**Pull request merged**